partiql-parser/src/lib.rs

```diff
@@ -2,10 +2,54 @@
 
 //! Provides a parser for the [PartiQL][partiql] query language.
 //!
+//! # Usage
+//!
+//! An API to interact with PartiQL tokens is the [`mod@scanner`] module.
+//! The [`scanner()`] function creates a [`Scanner`](scanner::Scanner) instance
+//! that one can use to parse tokens incrementally from some input slice.
+//!
+//! ```
+//! use partiql_parser::prelude::*;
+//! use partiql_parser::scanner;
+//!
+//! fn main() -> ParserResult<()> {
+//!     use partiql_parser::scanner::Content::*;
+//!
+//!     let mut scanner = scanner("SELECT FROM");
+//!     let first = scanner.next_token()?;
+//!
+//!     // get the parsed variant of the token
+//!     match first.content() {
+//!         Keyword(kw) => assert_eq!("SELECT", kw),
+//!     }
+//!     // the entire text of a token can be fetched--which looks the roughly the
+//!     // same for a keyword.
+//!     assert_eq!("SELECT", first.text());
```

Comment:
was expecting `as_str()` here

Reply:
SymbolToken can *not* have a str, right (`$0`)? so maybe that's an important difference - that `text()` might not be meaningful. Whereas here it seems obvious that the keyword can be "viewed as a str".

I see your pov. though, and I agree it makes sense.

Reply:
BTW, not that this is canonical or anything, but I've been more or less inspired by:

https://rust-lang.github.io/api-guidelines/naming.html

I think what I have here is more like a *getter* versus a *conversion*:

https://rust-lang.github.io/api-guidelines/naming.html#getter-names-follow-rust-convention-c-getter

>The difference between getters and conversions (C-CONV) can be subtle and is not always clear-cut. For example `TempDir::path` can be understood as a getter for the filesystem path of the temporary directory, while `TempDir::into_path` is a conversion that transfers responsibility for deleting the temporary directory to the caller. Since `path` is a getter, it would not be correct to call it `get_path` or `as_path`.